core/sparse_array.cpp

```diff
@@ -46,6 +46,25 @@ SparseArray::SparseArray(unsigned int p_nrows, unsigned int p_ncols)
 	
 	row_offsets_[nrows_set_] = 0;
 	finished_ = false;
+
+    //Allocate space for 2-D vectors
+    //Columns and Distances are now ragged 2-D vectors with each row corresponding to one focal individual. This negates the need for row offsets.
```

Comment:
Is this a change we want to make only in the parallel case, or a change that will be made for the single-threaded case also?  What are the performance implications of it for the single-threaded case?  Presumably negative; but how negative?

Reply:
This question remains important and unresolved.  If this new ragged-row design for SparseArray is significantly slower than the old design in the non-parallel case – which would not surprise me at all – then I think it would be justified to actually have two separate implementations of this class, one used when serial, the other used when parallel.  It is not a large class, and the distinction would be completely hidden behind a shared external API, so the messiness of it would be minimal.  We need speed testing to determine this.  This decision can/should be pended until the redesign of SparseArray is complete, though, and it passes the necessary validation tests.  Then we should test the speed impact for the serial case, and if it is significant, make the code duplication.